src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -372,7 +368,13 @@ export class UserService {
   }
 
   async activateUser(user?: User): Promise<void> {
-    await this.userRepo.activateUser(user);
+    // retry (in case of ref conflict)
+    user = await Util.retry(async () => {
+      user.ref = await this.getNextRef();
+      return this.userRepo.save(user);
+    }, 3);
+    await this.userRepo.activateUser(user, user.ref);
+    await this.userDataRepo.activateUserData(user.userData);
```

Comment:
Do not overwrite the ref. Save/activate at once.

```ts
// retry (in case of ref conflict)
await Util.retry(async () => {
  const ref = user.ref ?? await this.getNextRef();
  await this.userRepo.activateUser(user, ref);
}, 3);

if (user.userData.status === UserDataStatus.NA)
  await this.userDataRepo.activateUserData(user.userData);